src/API/Run.hs

```diff
@@ -10,49 +10,80 @@ Holds the routines for parsing and interpreting a BoGL Prelude and Game file.
 
 module API.Run (_runCodeWithCommands) where
 
-import API.JSONData
+import API.JSONData (SpielResponse(..), SpielCommand(..), SpielResponses, spielParseError)
 import Parser.Parser
 import Language.Syntax
 import Language.Types
 import Text.Parsec.Pos
-import Text.Parsec (errorPos)
-import Text.Parsec.Error
 
 import Typechecker.Typechecker
-import Runtime.Eval
+import Typechecker.Monad (Env(types))
+import Runtime.Eval (runWithBuffer, bindings_)
 import Runtime.Values
-import Runtime.Monad
+import Runtime.Monad (Buffer, emptyEnv)
+
+import Control.Monad(liftM, ap)
 
 -- | Runs BoGL code from raw text with the given commands
 -- utilizes parsePreludeAndGameText to parse the code directly,
 -- without reading it from a file first
```

Comment:
this function was unnecessarily in the IO monad! I pulled it out so that it is easier to test.